mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -242,9 +316,9 @@ class GeneralizedLinearRegression @Since("2.0.0") (@Since("2.0.0") override val
   def setLinkPredictionCol(value: String): this.type = set(linkPredictionCol, value)
 
   override protected def train(dataset: Dataset[_]): GeneralizedLinearRegressionModel = {
-    val familyObj = Family.fromName($(family))
-    val linkObj = if (isDefined(link)) {
-      Link.fromName($(link))
+    val familyObj = Family.fromParams(this)
+    val linkObj = if (isDefined(link) || isDefined(linkPower)) {
+      Link.fromParams(this)
     } else {
       familyObj.defaultLink
     }
```

Comment:
This code snippets are used multiple places, could we wrapper them up as a function?
```
def getFamilyAndLinkObj(params: GeneralizedLinearRegressionBase): (Family, Link) = {
......
}
```